### PR TITLE
Update deployment logic for settings changes

### PR DESCRIPTION
**Pull request by bdunne**
- Sync up deployments from the running bot
- Change the settings location to come from a secret now that the defaults are checked in and only secret values are stored in the yml

### CODE REVIEW

**Review by Fryguy — APPROVED** [review]

File: templates/bot.yaml [hunk 1]
```diff
@@ -93,76 +93,89 @@ objects:
...
+          - containerPort: 5432
+            protocol: TCP
+          readinessProbe:
+            failureThreshold: 3
+            initialDelaySeconds: 60
+            periodSeconds: 10
+            successThreshold: 1
+            tcpSocket:
+              port: 5432
+            timeoutSeconds: 1
```
Comment (Fryguy) [inline templates/bot.yaml:147]: Original code before the reorder was:

```suggestion
           readinessProbe:
              initialDelaySeconds: 60
              tcpSocket:
                port: 5432
```

File: templates/bot.yaml [hunk 2]
```diff
@@ -93,76 +93,89 @@ objects:
...
-            value: '1000'
+            value: "1000"
           - name: POSTGRESQL_SHARED_BUFFERS
             value: 1GB
+          image: docker.io/manageiq/postgresql:10
+          imagePullPolicy: IfNotPresent
+          name: postgresql
+          ports:
+          - containerPort: 5432
+            protocol: TCP
```
Comment (Fryguy) [inline templates/bot.yaml:139]: Technically the protocol was added

File: templates/bot.yaml [hunk 3]
```diff
@@ -93,76 +93,86 @@ objects:
...
+          name: pgsql-data-permission-fix
+          volumeMounts:
+          - mountPath: /var/lib/pgsql/data
+            name: pgdb-volume
+        volumes:
+        - name: pgdb-volume
+          persistentVolumeClaim:
+            claimName: bot-postgres-pvc
+        - configMap:
+            defaultMode: 420
```
Comment (Fryguy) [inline templates/bot.yaml:173]: All of these defaultMode look new, so I think we can drop them.

File: templates/bot.yaml [hunk 4]
```diff
@@ -180,44 +190,48 @@ objects:
...
-            claimName: bot-redis-pvc
         containers:
-        - name: redis
-          image: "docker.io/library/redis:5.0"
+        - image: docker.io/library/redis:5.0
+          imagePullPolicy: IfNotPresent
+          name: redis
           ports:
           - containerPort: 6379
+            protocol: TCP
```
Comment (Fryguy) [inline templates/bot.yaml:216]: Anoher TCP we can drop

File: templates/bot.yaml (hunk 2 above)
Comment (bdunne) [reply templates/bot.yaml:139]: I think we should keep these

**Pull request merged**